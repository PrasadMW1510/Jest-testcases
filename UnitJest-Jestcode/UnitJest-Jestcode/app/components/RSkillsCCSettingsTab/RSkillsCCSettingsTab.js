/**
 *
 * RSkillsCCSettingsTab
 *
 */
import PropTypes from 'prop-types';
import React, { Fragment } from 'react';

import LoadingBar from 'components/LoadingBar';
import ProgramSettingsButtons from 'components/ProgramSettingsButtons';
import {
  RSKILLSCC_SETTINGS_CHECKBOX_STATUS_MAP,
  RSKILLSCC_SETTING_GRADING_RUBRIC_FOUR_PT,
  RSKILLSCC_SETTING_GRADING_RUBRIC_SIX_PT,
} from 'components/RSkillsCCSetting/constants';
import SettingsFourStateCheckbox from 'components/SettingsFourStateCheckbox';
import { COHORT_TYPE } from 'containers/App/constants';

import './RSkillsCCSettingsTab.scss';

class RSkillsCCSettingsTab extends React.Component {
  constructor(props) {
    super(props);
    const settingsOnScreen = props.immProgramSettings.toJS();
    this.state = {
      settingsOnScreen,
      tabHasNoUnsavedChanges: true,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.immProgramSettings !== this.props.immProgramSettings) {
      this.setState({
        settingsOnScreen: nextProps.immProgramSettings.toJS(),
        tabHasNoUnsavedChanges: true,
      });
    }

    if (
      nextProps.immDefaultProgramSettings !== this.props.immDefaultProgramSettings &&
      nextProps.immDefaultProgramSettings.size > 0
    ) {
      // props.immDefaultProgramSettings will change as a result of the user clicking restore default.
      this.handleRestoreDefault(nextProps.immDefaultProgramSettings.toJS());
    }
  }

  englishLanguageLearnerAudioDirectionsMeta = [
    { id: 0, name: 'None' },
    { id: 2, name: 'Haitian Creole' },
    { id: 1, name: 'Spanish' },
    { id: 5, name: 'Hmong' },
    { id: 3, name: 'Cantonese' },
    { id: 4, name: 'Vietnamese' },
  ];

  createRSkillsCCSettingsPayload = redirectToRoster => {
    const { effectiveCohortObject } = this.props;
    const {
      audio_instructions,
      show_correct_incorrect,
      include_open_response,
      include_writing_prompts,
      ell_audio_instructions,
      writing_prompt_grading,
    } = this.state.settingsOnScreen;

    const result = {
      redirectToRoster,
      postPayload: {
        output: {
          output_data: {
            cohort_type: effectiveCohortObject.cohortType.toUpperCase(),
            cohort_id: effectiveCohortObject.id,
            settings: {
              audio_instructions: audio_instructions[0],
              show_correct_incorrect: show_correct_incorrect[0],
              include_open_response: include_open_response[0],
              include_writing_prompts: include_writing_prompts[0],
              ell_audio_instructions: ell_audio_instructions[0],
              writing_prompt_grading: {
                grading_rubric: writing_prompt_grading[0].grading_rubric[0],
              },
            },
          },
        },
      },
    };
    return result;
  };

  fromApiToCheckboxValue = apiValue => (apiValue ? apiValue[0] : '0');

  handleChange = e => {
    const targetFieldName = e.target.name;
    let newValue = e.target.value;
    const isChecked = e.target.checked;
    const newState = {
      settingsOnScreen: this.state.settingsOnScreen,
      tabHasNoUnsavedChanges: false,
    };
    if (e.target.type === 'checkbox' && isChecked === true) {
      newValue = 1;
    } else if (e.target.type === 'checkbox' && isChecked === false) {
      newValue = 0;
    }
    if (e.target.name === 'writing_rubric') {
      // writing rubric is nested one deep, so cannot use targetFieldName
      newState.settingsOnScreen.writing_prompt_grading[0].grading_rubric = [String(newValue)];
    } else {
      newState.settingsOnScreen[targetFieldName] = [String(newValue)];
    }
    this.setState(newState);
    this.props.handleIsolateTab();
  };

  /**
   * this gets called, by way of componentWillReceiveProps, after a user has clicked the RestoreDefaults button
   */
  handleRestoreDefault = defaultSettings => {
    this.setState({
      settingsOnScreen: defaultSettings,
      tabHasNoUnsavedChanges: false,
    });
    this.props.handleIsolateTab();
  };

  handleRestoreDefaultClick = () => {
    this.props.handleRSkillsCCGetDefaultSettings();
  };

  handleSetInitialValues = () => {
    const settingsOnScreen = this.props.immProgramSettings.toJS();
    this.setState({
      settingsOnScreen,
      tabHasNoUnsavedChanges: true,
    });
    this.props.handleTabReset();
  };

  handleSubmit = e => {
    e.preventDefault();
    const noRedirect = false;
    this.props.handleSave(this.createRSkillsCCSettingsPayload(noRedirect));
    this.props.handleTabReset();
    this.hideSaveOptions();
  };

  handleSaveAndReturn = () => {
    const redirect = true;
    this.props.handleSave(this.createRSkillsCCSettingsPayload(redirect));
  };

  hideSaveOptions = () => {
    this.setState({ tabHasNoUnsavedChanges: true });
  };

  writingPromptRubricChecked = rubricVal => {
    let result = false;
    try {
      result =
        Number(this.state.settingsOnScreen.writing_prompt_grading[0].grading_rubric[0]) ===
        Number(rubricVal);
    } catch (err) {
      result = false;
    }
    return result;
  };

  renderCheckboxLabelRSkillCCSetting = (label, apiProperty) => {
    const checkboxValue = this.fromApiToCheckboxValue(this.state.settingsOnScreen[apiProperty]);
    return (
      <div className="rskillscc-program-settings__setting">
        <SettingsFourStateCheckbox
          checkboxName={apiProperty}
          checkboxText={label}
          currentCheckboxValue={checkboxValue}
          handleChangeCheckboxValue={this.handleChange}
          statusMap={RSKILLSCC_SETTINGS_CHECKBOX_STATUS_MAP}
        />
      </div>
    );
  };

  renderTestExperience = () => (
    <div className="rskillscc-program-settings__test-experience-settings">
      <span className="rskillscc-program-settings__caption">Test Experience</span>
      {this.renderCheckboxLabelRSkillCCSetting('Include audio directions', 'audio_instructions')}
      {this.renderCheckboxLabelRSkillCCSetting(
        'Show correct and incorrect answers',
        'show_correct_incorrect'
      )}
      {this.renderCheckboxLabelRSkillCCSetting(
        'Include constructed response',
        'include_open_response'
      )}
      {this.renderCheckboxLabelRSkillCCSetting(
        'Include extended constructed response',
        'include_writing_prompts'
      )}
    </div>
  );

  renderEnglishLanguageLearner = () => (
    <div className="rskillscc-program-settings__ell-settings">
      <span className="rskillscc-program-settings__caption">
        English-Lanaguage Learner Audio Directions
      </span>
      {this.englishLanguageLearnerAudioDirectionsMeta.map(option => (
        <div className="rskillscc-program-settings__ell-audio-instructions" key={option.name}>
          <label htmlFor={option.name} className="rskillscc-program-settings__radio-label">
            <input
              type="radio"
              className="rskillscc-program-settings__radio"
              name="ell_audio_instructions"
              onChange={this.handleChange}
              id={option.name}
              key={option.name}
              value={option.id}
              checked={
                this.state.settingsOnScreen.ell_audio_instructions &&
                Number(this.state.settingsOnScreen.ell_audio_instructions[0]) === Number(option.id)
              }
            />
            {option.name}
          </label>
        </div>
      ))}
    </div>
  );

  renderWritingPromptSettings = () => {
    const disableRadio = this.props.effectiveCohortObject.cohortType !== COHORT_TYPE.District;
    const disabledCursorCSS = disableRadio ? 'rskillscc-program-settings__cursor--default' : '';
    const labelCSSClass = `rskillscc-program-settings__radio-label ${disabledCursorCSS}`;
    const radioCSSClass = `rskillscc-program-settings__radio  ${disabledCursorCSS}`;
    return (
      <div className="rskillscc-program-settings__writing-prompt-settings">
        <span className="rskillscc-program-settings__caption">Writing Prompt Settings</span>
        <div className="rskillscc-program-settings__writing-prompt-settings-instructions">
          Choose either a 4-point or 6-point rubric to grade rSkills Tests: College & Career Writing
          Prompts. To preview the rubrics for Open Response questions or Writing Prompts, see the
          SAM Resources.
        </div>
        <div className="rskillscc-program-settings__writing-prompt-settings-rubric-container">
          <div className="rskillscc-program-settings__rubric-options" key="rubric_four_id">
            <label htmlFor="rubric_four_id" className={labelCSSClass}>
              <input
                type="radio"
                className={radioCSSClass}
                name="writing_rubric"
                onChange={this.handleChange}
                id="rubric_four_id"
                key="rubric_four_name"
                value={RSKILLSCC_SETTING_GRADING_RUBRIC_FOUR_PT}
                checked={this.writingPromptRubricChecked(RSKILLSCC_SETTING_GRADING_RUBRIC_FOUR_PT)}
                disabled={disableRadio}
              />
              4 point rubric
            </label>
          </div>
          <div className="rskillscc-program-settings__rubric-options" key="rubric_six_id">
            <label htmlFor="rubric_six_id" className={labelCSSClass}>
              <input
                type="radio"
                className={radioCSSClass}
                name="writing_rubric"
                onChange={this.handleChange}
                id="rubric_six_id"
                key="rubric_six_name"
                value={RSKILLSCC_SETTING_GRADING_RUBRIC_SIX_PT}
                checked={this.writingPromptRubricChecked(RSKILLSCC_SETTING_GRADING_RUBRIC_SIX_PT)}
                disabled={disableRadio}
              />
              6 point rubric
            </label>
          </div>
        </div>
      </div>
    );
  };
  renderSettings = () => (
    <Fragment>
      <div className="rskillscc-settings-tab">
        <div className="rskillscc-program-settings__section--full-width rskillscc-program-settings__section--top">
          <div className="rskillscc-program-settings__section--top-inputs">
            {this.renderTestExperience()}
            {this.renderEnglishLanguageLearner()}
          </div>
        </div>
        <div className="rskillscc-program-settings__section--full-width">
          {this.renderWritingPromptSettings()}
        </div>
      </div>
      <div className="rskillscc-program-settings__program-setting-buttons-container">
        <ProgramSettingsButtons
          restoreDefaultHandler={this.handleRestoreDefaultClick}
          saveAndReturnHandler={this.handleSaveAndReturn}
          setInitialValuesHandler={this.handleSetInitialValues}
          stateResult={!this.props.isTabIsolated}
          suppressSaveAndReturnRedirect
        />
      </div>
    </Fragment>
  );

  render() {
    if (this.props.isLoading || this.props.isDefaultProgramSettingsLoading) {
      return (
        <div className="rskillscc-settings-tab">
          <LoadingBar />
        </div>
      );
    }
    return <form onSubmit={this.handleSubmit}>{this.renderSettings()}</form>;
  }
}

RSkillsCCSettingsTab.defaultProps = {
  isDefaultProgramSettingsLoading: false,
  isLoading: false,
};

RSkillsCCSettingsTab.propTypes = {
  effectiveCohortObject: PropTypes.object.isRequired,
  handleIsolateTab: PropTypes.func.isRequired,
  handleRSkillsCCGetDefaultSettings: PropTypes.func.isRequired,
  handleTabReset: PropTypes.func.isRequired,
  handleSave: PropTypes.func.isRequired,
  isDefaultProgramSettingsLoading: PropTypes.bool,
  isLoading: PropTypes.bool,
  immDefaultProgramSettings: PropTypes.any,
  immProgramSettings: PropTypes.any,
  isTabIsolated: PropTypes.bool.isRequired,
};

export default RSkillsCCSettingsTab;
