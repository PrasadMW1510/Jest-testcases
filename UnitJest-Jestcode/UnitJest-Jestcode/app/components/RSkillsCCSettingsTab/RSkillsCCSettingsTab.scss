.rskillscc-settings-tab {
  background-color: #fff;
  display: inline-block;
  font-family: sans-serif;
  font-size: 9pt;
  height: 395px;
  margin-left: 1%;
  position: relative;
  width: 98%;
}

.rskillscc-program-settings__program-setting-buttons-container {
  bottom: 5px;
  position: absolute;
  width: 99%;
}

.rskillscc-program-settings__test-experience-settings {
  border-right: 1px solid #000;
  float: left;
  width: 40%;
}

.rskillscc-program-settings__ell-settings {
  float: left;
  width: 60%;
}

.rskillscc-program-settings__writing-prompt-settings {
  border: 1px solid #000;
  float: left;
  width: 100%;
}

.rskillscc-program-settings__caption {
  font-weight: bold;
  vertical-align: top;
  padding: 3px 0 0 10px;
  background-color: #ffd400;
  width: 100%;
  height: 25px;
  display: inline-block;
  margin-top: 0%;
  z-index: +1;
}

.rskillscc-program-settings__setting {
  padding: 3px 3px 3px 8px;
}

.rskillscc-program-settings__radio {
  margin-right: 4px;
  cursor: pointer;
  margin-top: -1px;
  vertical-align: middle;
}

.rskillscc-program-settings__radio-label {
  font-family: sans-serif;
  cursor: pointer;
}

.rskillscc-program-settings__cursor--default {
  cursor: default;
}

.rskillscc-program-settings__ell-audio-instructions {
  float: left;
  width: 50%;
  height: 25%;
  margin: 0;
  margin-top: 2%;
  font-size: 12px;
  font-weight: normal;
  padding-left: 10px;
}

.rskillscc-program-settings__rubric-options {
  float: left;
  width: 50%;
  height: 25%;
  margin: 0;
  margin-top: 2%;
  font-size: 12px;
  font-weight: normal;
  padding-left: 10px;
  padding-bottom: 50px;
}

.rskillscc-program-settings__section--full-width {
  width: 100%;
}

.rskillscc-program-settings__section--top {
  height: 40%;
}

.rskillscc-program-settings__section--top-inputs {
  border: 1px solid #000;
  float: left;
}

.rskillscc-program-settings__section--half-height {
  height: 50%;
}

.rskillscc-program-settings__writing-prompt-settings-instructions {
  background-color: #ffe279;
  padding: 3px 10px 10px;
}
