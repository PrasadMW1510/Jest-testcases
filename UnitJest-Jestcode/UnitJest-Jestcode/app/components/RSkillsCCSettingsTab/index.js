/**
 *
 * RSkillsCCSettingsTab
 *
 */

export { default } from './RSkillsCCSettingsTab';
