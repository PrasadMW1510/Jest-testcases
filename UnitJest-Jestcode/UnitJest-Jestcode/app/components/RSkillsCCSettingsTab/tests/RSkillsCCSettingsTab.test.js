import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { fromJS } from 'immutable';
import RSkillsCCSettingsTab from '../index';

describe('<RSkillsCCSettingsTab />', () => {
  let wrapper = null;
  const mockStudentCohortObject = { cohortType: 'Student', id: 'guidStudent54321' };
  const mockTabReset = jest.fn();
  const mockIsolateTab = jest.fn();
  const mockHandleSave = jest.fn();
  const mockHandleGetDefaultSettings = jest.fn();
  const fakeTrueCheckboxEvent = {
    target: {
      checked: true,
      type: 'checkbox',
    },
  };
  const fakeFalseCheckboxEvent = {
    target: {
      checked: false,
      type: 'checkbox',
    },
  };
  const mockDefaultProgramSettingsEmpty = fromJS({});
  const mockDefaultProgramSettings = fromJS({
    ell_audio_instructions: ['1'],
    audio_instructions: ['1'],
    show_correct_incorrect: ['1'],
    include_open_response: ['1'],
    include_writing_prompts: ['1'],
    writing_prompt_grading: [{ active: ['true'], grading_rubric: ['0'] }],
  });
  const mockProgramSettings = fromJS({
    ell_audio_instructions: ['1'],
    audio_instructions: ['1'],
    show_correct_incorrect: ['1'],
    include_open_response: ['1'],
    include_writing_prompts: ['1'],
    writing_prompt_grading: [{ active: ['true'], grading_rubric: ['0'] }],
  });
  const mockNewProgramSettings = fromJS({
    ell_audio_instructions: ['1'],
    audio_instructions: ['1'],
    show_correct_incorrect: ['0'],
    include_open_response: ['0'],
    include_writing_prompts: ['0'],
    writing_prompt_grading: [{ active: ['true'], grading_rubric: ['0'] }],
  });
  beforeEach(() => {
    wrapper = shallow(
      <RSkillsCCSettingsTab
        effectiveCohortObject={mockStudentCohortObject}
        handleIsolateTab={mockIsolateTab}
        handleRSkillsCCGetDefaultSettings={mockHandleGetDefaultSettings}
        handleSave={mockHandleSave}
        handleTabReset={mockTabReset}
        isTabIsolated={false}
        immProgramSettings={mockProgramSettings}
        immDefaultProgramSettings={mockDefaultProgramSettingsEmpty}
      />
    );
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render as expected', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should handle the restore default', () => {
    wrapper.instance().handleRestoreDefaultClick();
    expect(mockHandleGetDefaultSettings).toHaveBeenCalled();
  });

  it('should handle the submit', () => {
    const e = {
      preventDefault: () => {},
    };
    wrapper.instance().handleSubmit(e);
    expect(mockHandleSave).toHaveBeenCalled();
    expect(mockTabReset).toHaveBeenCalled();
  });

  it('should handle the saveAndReturn', () => {
    wrapper.instance().handleSaveAndReturn();
    expect(mockHandleSave).toHaveBeenCalled();
  });

  it('should handle new programsettings that are same as initial.', () => {
    expect(wrapper.state('tabHasNoUnsavedChanges')).toBeTruthy();
    wrapper.setState({ tabHasNoUnsavedChanges: false });
    expect(wrapper.state('tabHasNoUnsavedChanges')).toBeFalsy();
    wrapper.setProps({ immProgramSettings: mockProgramSettings });
    expect(wrapper.state('tabHasNoUnsavedChanges')).toBeFalsy();
  });

  it('should handle the changed settings props.', () => {
    expect(wrapper.state('tabHasNoUnsavedChanges')).toBeTruthy();
    wrapper.setState({ tabHasNoUnsavedChanges: false });
    expect(wrapper.state('tabHasNoUnsavedChanges')).toBeFalsy();
    wrapper.setProps({ immProgramSettings: mockNewProgramSettings });
    expect(wrapper.state('tabHasNoUnsavedChanges')).toBeTruthy();
  });

  it('should handle the changed default settings props.', () => {
    expect(wrapper.state('tabHasNoUnsavedChanges')).toBeTruthy();
    wrapper.setState({ tabHasNoUnsavedChanges: false });
    expect(wrapper.state('tabHasNoUnsavedChanges')).toBeFalsy();
    expect(mockIsolateTab).not.toHaveBeenCalled();
    wrapper.setProps({ immDefaultProgramSettings: mockDefaultProgramSettings });
    expect(mockIsolateTab).toHaveBeenCalled();
  });

  it('should handle the setInitialValues', () => {
    expect(mockTabReset).not.toHaveBeenCalled();
    wrapper.instance().handleSetInitialValues();
    expect(mockTabReset).toHaveBeenCalled();
  });

  it('should call isolateTab function when user checks a box', () => {
    expect(mockIsolateTab).not.toHaveBeenCalled();
    const correctAndIncorrectAnswerCheckbox = wrapper.find(
      'SettingsFourStateCheckbox[checkboxText="Show correct and incorrect answers"]'
    );
    correctAndIncorrectAnswerCheckbox.prop('handleChangeCheckboxValue')(fakeTrueCheckboxEvent);
    expect(mockIsolateTab).toHaveBeenCalled();
  });

  it('should call isolateTab function when user unchecks a box', () => {
    expect(mockIsolateTab).not.toHaveBeenCalled();
    const correctAndIncorrectAnswerCheckbox = wrapper.find(
      'SettingsFourStateCheckbox[checkboxText="Show correct and incorrect answers"]'
    );
    correctAndIncorrectAnswerCheckbox.prop('handleChangeCheckboxValue')(fakeTrueCheckboxEvent);
    correctAndIncorrectAnswerCheckbox.prop('handleChangeCheckboxValue')(fakeFalseCheckboxEvent);
    expect(mockIsolateTab).toHaveBeenCalled();
  });

  it('should call isolateTab function when user updates writing rubric ', () => {
    expect(mockIsolateTab).not.toHaveBeenCalled();
    const rubricSix = wrapper.find('#rubric_six_id');
    rubricSix.simulate('change', {
      target: { value: 'true', name: 'writing_rubric' },
    });
    expect(mockIsolateTab).toHaveBeenCalled();
  });

  describe('when isLoading is true', () => {
    beforeEach(() => {
      wrapper = shallow(
        <RSkillsCCSettingsTab
          effectiveCohortObject={mockStudentCohortObject}
          handleIsolateTab={mockIsolateTab}
          handleRSkillsCCGetDefaultSettings={mockHandleGetDefaultSettings}
          handleSave={mockHandleSave}
          handleTabReset={mockTabReset}
          isTabIsolated={false}
          immProgramSettings={mockProgramSettings}
          immDefaultProgramSettings={mockDefaultProgramSettingsEmpty}
          isLoading
        />
      );
    });

    it('should render a LoadingBar', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find('LoadingBar')).toHaveLength(1);
      wrapper.setProps({ isLoading: false });
      expect(wrapper.find('LoadingBar').exists()).toBeFalsy();
    });
  });
});
