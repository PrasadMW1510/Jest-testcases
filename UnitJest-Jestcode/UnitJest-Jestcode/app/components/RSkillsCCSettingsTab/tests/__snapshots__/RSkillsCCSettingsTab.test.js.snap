// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RSkillsCCSettingsTab /> should render as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <React.Fragment>
    <div
      className="rskillscc-settings-tab"
    >
      <div
        className="rskillscc-program-settings__section--full-width rskillscc-program-settings__section--top"
      >
        <div
          className="rskillscc-program-settings__section--top-inputs"
        >
          <div
            className="rskillscc-program-settings__test-experience-settings"
          >
            <span
              className="rskillscc-program-settings__caption"
            >
              Test Experience
            </span>
            <div
              className="rskillscc-program-settings__setting"
            >
              <SettingsFourStateCheckbox
                checkboxClass=""
                checkboxName="audio_instructions"
                checkboxText="Include audio directions"
                currentCheckboxValue="1"
                disabled={false}
                handleChangeCheckboxValue={[Function]}
                statusMap={
                  Object {
                    "checkedDisabled": Array [],
                    "checkedEnabled": Array [
                      "1",
                    ],
                    "mixedDisabled": Array [],
                    "mixedEnabled": Array [
                      "-2",
                    ],
                    "uncheckedDisabled": Array [],
                    "uncheckedEnabled": Array [
                      "0",
                    ],
                  }
                }
                textClass=""
              />
            </div>
            <div
              className="rskillscc-program-settings__setting"
            >
              <SettingsFourStateCheckbox
                checkboxClass=""
                checkboxName="show_correct_incorrect"
                checkboxText="Show correct and incorrect answers"
                currentCheckboxValue="1"
                disabled={false}
                handleChangeCheckboxValue={[Function]}
                statusMap={
                  Object {
                    "checkedDisabled": Array [],
                    "checkedEnabled": Array [
                      "1",
                    ],
                    "mixedDisabled": Array [],
                    "mixedEnabled": Array [
                      "-2",
                    ],
                    "uncheckedDisabled": Array [],
                    "uncheckedEnabled": Array [
                      "0",
                    ],
                  }
                }
                textClass=""
              />
            </div>
            <div
              className="rskillscc-program-settings__setting"
            >
              <SettingsFourStateCheckbox
                checkboxClass=""
                checkboxName="include_open_response"
                checkboxText="Include constructed response"
                currentCheckboxValue="1"
                disabled={false}
                handleChangeCheckboxValue={[Function]}
                statusMap={
                  Object {
                    "checkedDisabled": Array [],
                    "checkedEnabled": Array [
                      "1",
                    ],
                    "mixedDisabled": Array [],
                    "mixedEnabled": Array [
                      "-2",
                    ],
                    "uncheckedDisabled": Array [],
                    "uncheckedEnabled": Array [
                      "0",
                    ],
                  }
                }
                textClass=""
              />
            </div>
            <div
              className="rskillscc-program-settings__setting"
            >
              <SettingsFourStateCheckbox
                checkboxClass=""
                checkboxName="include_writing_prompts"
                checkboxText="Include extended constructed response"
                currentCheckboxValue="1"
                disabled={false}
                handleChangeCheckboxValue={[Function]}
                statusMap={
                  Object {
                    "checkedDisabled": Array [],
                    "checkedEnabled": Array [
                      "1",
                    ],
                    "mixedDisabled": Array [],
                    "mixedEnabled": Array [
                      "-2",
                    ],
                    "uncheckedDisabled": Array [],
                    "uncheckedEnabled": Array [
                      "0",
                    ],
                  }
                }
                textClass=""
              />
            </div>
          </div>
          <div
            className="rskillscc-program-settings__ell-settings"
          >
            <span
              className="rskillscc-program-settings__caption"
            >
              English-Lanaguage Learner Audio Directions
            </span>
            <div
              className="rskillscc-program-settings__ell-audio-instructions"
              key="None"
            >
              <label
                className="rskillscc-program-settings__radio-label"
                htmlFor="None"
              >
                <input
                  checked={false}
                  className="rskillscc-program-settings__radio"
                  id="None"
                  key="None"
                  name="ell_audio_instructions"
                  onChange={[Function]}
                  type="radio"
                  value={0}
                />
                None
              </label>
            </div>
            <div
              className="rskillscc-program-settings__ell-audio-instructions"
              key="Haitian Creole"
            >
              <label
                className="rskillscc-program-settings__radio-label"
                htmlFor="Haitian Creole"
              >
                <input
                  checked={false}
                  className="rskillscc-program-settings__radio"
                  id="Haitian Creole"
                  key="Haitian Creole"
                  name="ell_audio_instructions"
                  onChange={[Function]}
                  type="radio"
                  value={2}
                />
                Haitian Creole
              </label>
            </div>
            <div
              className="rskillscc-program-settings__ell-audio-instructions"
              key="Spanish"
            >
              <label
                className="rskillscc-program-settings__radio-label"
                htmlFor="Spanish"
              >
                <input
                  checked={true}
                  className="rskillscc-program-settings__radio"
                  id="Spanish"
                  key="Spanish"
                  name="ell_audio_instructions"
                  onChange={[Function]}
                  type="radio"
                  value={1}
                />
                Spanish
              </label>
            </div>
            <div
              className="rskillscc-program-settings__ell-audio-instructions"
              key="Hmong"
            >
              <label
                className="rskillscc-program-settings__radio-label"
                htmlFor="Hmong"
              >
                <input
                  checked={false}
                  className="rskillscc-program-settings__radio"
                  id="Hmong"
                  key="Hmong"
                  name="ell_audio_instructions"
                  onChange={[Function]}
                  type="radio"
                  value={5}
                />
                Hmong
              </label>
            </div>
            <div
              className="rskillscc-program-settings__ell-audio-instructions"
              key="Cantonese"
            >
              <label
                className="rskillscc-program-settings__radio-label"
                htmlFor="Cantonese"
              >
                <input
                  checked={false}
                  className="rskillscc-program-settings__radio"
                  id="Cantonese"
                  key="Cantonese"
                  name="ell_audio_instructions"
                  onChange={[Function]}
                  type="radio"
                  value={3}
                />
                Cantonese
              </label>
            </div>
            <div
              className="rskillscc-program-settings__ell-audio-instructions"
              key="Vietnamese"
            >
              <label
                className="rskillscc-program-settings__radio-label"
                htmlFor="Vietnamese"
              >
                <input
                  checked={false}
                  className="rskillscc-program-settings__radio"
                  id="Vietnamese"
                  key="Vietnamese"
                  name="ell_audio_instructions"
                  onChange={[Function]}
                  type="radio"
                  value={4}
                />
                Vietnamese
              </label>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rskillscc-program-settings__section--full-width"
      >
        <div
          className="rskillscc-program-settings__writing-prompt-settings"
        >
          <span
            className="rskillscc-program-settings__caption"
          >
            Writing Prompt Settings
          </span>
          <div
            className="rskillscc-program-settings__writing-prompt-settings-instructions"
          >
            Choose either a 4-point or 6-point rubric to grade rSkills Tests: College & Career Writing Prompts. To preview the rubrics for Open Response questions or Writing Prompts, see the SAM Resources.
          </div>
          <div
            className="rskillscc-program-settings__writing-prompt-settings-rubric-container"
          >
            <div
              className="rskillscc-program-settings__rubric-options"
              key="rubric_four_id"
            >
              <label
                className="rskillscc-program-settings__radio-label rskillscc-program-settings__cursor--default"
                htmlFor="rubric_four_id"
              >
                <input
                  checked={false}
                  className="rskillscc-program-settings__radio  rskillscc-program-settings__cursor--default"
                  disabled={true}
                  id="rubric_four_id"
                  key="rubric_four_name"
                  name="writing_rubric"
                  onChange={[Function]}
                  type="radio"
                  value={1}
                />
                4 point rubric
              </label>
            </div>
            <div
              className="rskillscc-program-settings__rubric-options"
              key="rubric_six_id"
            >
              <label
                className="rskillscc-program-settings__radio-label rskillscc-program-settings__cursor--default"
                htmlFor="rubric_six_id"
              >
                <input
                  checked={true}
                  className="rskillscc-program-settings__radio  rskillscc-program-settings__cursor--default"
                  disabled={true}
                  id="rubric_six_id"
                  key="rubric_six_name"
                  name="writing_rubric"
                  onChange={[Function]}
                  type="radio"
                  value={0}
                />
                6 point rubric
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="rskillscc-program-settings__program-setting-buttons-container"
    >
      <ProgramSettingsButtons
        restoreDefaultHandler={[Function]}
        saveAndReturnHandler={[Function]}
        setInitialValuesHandler={[Function]}
        showRestoreDefaults={true}
        stateResult={true}
        suppressSaveAndReturnRedirect={true}
      />
    </div>
  </React.Fragment>
</form>
`;

exports[`<RSkillsCCSettingsTab /> when isLoading is true should render a LoadingBar 1`] = `
<div
  className="rskillscc-settings-tab"
>
  <LoadingBar />
</div>
`;
